### PR TITLE
Always catch runtime exceptions from serializers and deserializers

### PR DESCRIPTION
**Pull request by dstepanov**
Fixes https://github.com/micronaut-projects/micronaut-serialization/issues/878

### CODE REVIEW

**Review by yawkat — APPROVED** [review]

File: serde-support/src/main/java/io/micronaut/serde/support/deserializers/ErrorCatchingDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import java.io.IOException;
+
+/**
+ * The error catching {@link Deserializer}.
+ *
+ * @param <T> The deserializer type
+ * @author Denis Stepanov
+ */
+@Internal
+class ErrorCatchingDeserializer<T> implements Deserializer<T> {
```
Comment (graemerocher) [inline serde-support/src/main/java/io/micronaut/serde/support/deserializers/ErrorCatchingDeserializer.java:34]: can this be `final`?

File: serde-support/src/main/java/io/micronaut/serde/support/serializers/ErrorCatchingSerializer.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import java.io.IOException;
+
+/**
+ * The errors catching {@link Serializer}.
+ *
+ * @param <T> The serializer type
+ * @author Denis Stepanov
+ */
+@Internal
+class ErrorCatchingSerializer<T> implements Serializer<T> {
```
Comment (graemerocher) [inline serde-support/src/main/java/io/micronaut/serde/support/serializers/ErrorCatchingSerializer.java:34]: can this be `final`?

**Review by graemerocher — APPROVED** [review]

File: serde-support/src/main/java/io/micronaut/serde/support/deserializers/ErrorCatchingDeserializer.java (hunk 1 above)
Comment (dstepanov) [reply serde-support/src/main/java/io/micronaut/serde/support/deserializers/ErrorCatchingDeserializer.java:34]: It’s inherited

File: serde-support/src/main/java/io/micronaut/serde/support/serializers/ErrorCatchingSerializer.java (hunk 2 above)
Comment (dstepanov) [reply serde-support/src/main/java/io/micronaut/serde/support/serializers/ErrorCatchingSerializer.java:34]: It’s inherited

**Pull request merged**